core/src/stored/sd_stats.cc

```diff
@@ -72,53 +72,54 @@ struct device_statistic {
   uint64_t VolCatBlocks{0};
 };
 
-struct device_tapealert {
-  dlink link;
+struct device_tapealert_item {
+  dlink<device_tapealert_item> link;
   utime_t timestamp{0};
   uint64_t flags{0};
 };
 
-struct device_statistics {
-  dlink link;
+struct device_statistics_t {
+  dlink<device_statistics_t> link;
```

Comment:
i really don't like naming type `whatever_t`, as this makes it look like one of the stdlib type aliases. However, that's probably just a matter of taste.